components/molecules/AuthSection/auth-section.tsx

```diff
@@ -18,10 +18,50 @@ import DropdownList from "../DropdownList/dropdown-list";
 import Text from "components/atoms/Typography/text";
 import GitHubIcon from "img/icons/github-icon.svg";
 import Icon from "components/atoms/Icon/icon";
+import { Popover, PopoverContent, PopoverTrigger } from "../Popover/popover";
+import { IoNotifications } from "react-icons/io5";
+import NotificationCard from "components/atoms/NotificationsCard/notification-card";
+import { useEffect, useState } from "react";
+import { authSession } from "lib/hooks/authSession";
+import { Spinner } from "components/atoms/SpinLoader/spin-loader";
 
 const AuthSection: React.FC = ({}) => {
-  const { signIn, signOut, user } = useSupabaseAuth();
+  const { signIn, signOut, user, sessionToken } = useSupabaseAuth();
   const { onboarded } = useSession();
+  const [notifications, setNotifications] = useState<DbNotification[]>([]);
+  const [loading, setLoading] = useState(false);
+  const [userInfo, setUserInfo] = useState<DbUser | undefined>(undefined);
+
+  // Fetch user notifications
+  const fetchNotifications = async () => {
+    if (userInfo && userInfo.notification_count > 0) {
+      setLoading(true);
+      const req = await fetch(`${process.env.NEXT_PUBLIC_API_URL}/user/notifications`, {
+        headers: {
+          accept: "application/json",
+          Authorization: `Bearer ${sessionToken}`
+        }
+      });
+      setLoading(false);
+      if (req.ok) {
+        const notifications = await req.json();
+        setNotifications(notifications.data as DbNotification[]);
+      }
+    } else {
+      return;
+    }
+  };
+
+  useEffect(() => {
+    const getUser = async () => {
+      const response = await authSession();
```

Comment:
nit: We should refactor the `useSession` and `authSession` hooks into one hook. Not needed for this PR